### DISCUSSION TITLE
Online Assisted-Installer service for OKD

### DISCUSSION THREAD

**Discussion by lveyde** [okd-project/okd#1112] [Ideas]
OpenShift provides an easy way to create a new OpenShift cluster using an Assisted-Installer service, providing a readily available service on console.openshift.com.

Following latest patches by @vrutkovs to the assisted-service it's now supports OKD.
However right now it's not as easy to use as the one for the OpenShift, since the user needs to manually build and integrate together all service components, build a FCOS based discovery image etc. etc. to basically get to the same starting point OpenShift already provides out of the box.

Thus it would be great to have a similar sort of service for OKD, so one could create a new OKD cluster just as easily.

That may probably require some changes i.e. using Google/GitHub OpenID for authentication.

Also as discussed at the OKD-WG meeting, it may be a good idea to additionally create an appliance sort of image that would provide this same service for offline installations, in an easily downloadable package, i.e. VM image.

**Comment by sandrobonazzola** [comment]
From an OKD user perspective this is a great idea.

**Comment by romfreiman** [comment]
For appliance like experience - you can bake the podman version into a livecd with small service file to auto start it upon boot.
For a hosted version - I think the gap is where to run it? who gonna host it? and maintain?

**Reply by vrutkovs** [reply]
Summarizing my concerns from DMs and WG discussion:
* service ownership (who's going to maintain it)
* user data handling requires ToS and authentication
* most OKD installations are disconnected

**Reply by celebdor** [reply]
Since this is based on Fedora CoreOS, I wonder if the Fedora CPE team would like to host it.

**Reply by lveyde** [reply]
@vrutkovs Regarding appliance and offline installations - it looks that currently an AWS S3 service is required, so it's not really possible to use AI flow for offline installations, or am I'm missing something?

**Reply by LorbusChris** [reply]
@celebdor I like that idea, are we all in agreement that we should pursue that avenue?

There's a guide for working with CPE here: https://docs.fedoraproject.org/en-US/cpe/working_with_us/

**Comment by romfreiman** [comment]
In an offline mode, s3 is replaced with local file system.

**Reply by lveyde** [reply]
OK, thanks!
